src/compiler/checker.ts

```diff
@@ -28515,11 +28515,49 @@ namespace ts {
             }
         }
 
+        function getNonSimpleParameters(parameters: ReadonlyArray<ParameterDeclaration>): ReadonlyArray<ParameterDeclaration> {
+            // ECMA-262 14.1.13
+            if (parameters.length === 0) return [];
+
+            const last = lastOrUndefined(parameters);
+            if (last && isRestParameter(last)) return [last];

```

Comment:
The presence of a rest parameter likely shouldn't preclude other parameters from being non-simple. Since an error is issued on each non-simple parameter, I think the restiness check aughta get moved into the filter below (this way fixing one error on the rest doesn't just move the error to the initialized parameter before it, for example).

A test of this would be good, too:
```ts
function foo(a = 3, ... b) {
  "use strict";
}
```